elastic/kibana
# pass elasticsearch response to browser as a stream

Remove decoding and re-encoding of the ES-response from Kibana-server.  Resolves https://github.com/elastic/kibana/issues/189640


- Only works with bfetch disabled (bfetch:disable advanced setting set to true, which is the default)
- only works with async esql and async dsl search strategies
- In above cases it does not parse ES response server-side, but forwards the Readable stream from the ES JS client to the HTTP response
- Reads ID (x-elasticsearch-async-id) and completion status (x-elasticsearch-async-is-running) from headers so that parsing the body can happen on the client rather than both on the client & server

How to test:
 when using ese or async esql search strategies (discover, lens, maps, ....) you should see in the network tab of developer tools that the responses to your requests come directly from es and they should be gzipped
<img width="697" alt="Screenshot 2024-10-07 at 11 41 11" src="https://github.com/user-attachments/assets/982a5282-300a-4a91-a962-ac5d9917f165">

- functionality wise nothing should change
